c7n/resources/s3.py

```diff
@@ -2795,8 +2802,14 @@ def process_bucket(self, b):
             if e.response['Error']['Code'] != 'ServerSideEncryptionConfigurationNotFoundError':
                 raise
 
-        for sse in rules:
-            if self.filter_bucket(b, sse):
+        # default `state` to True as previous impl assumed state == True
+        # to preserve backwards compatibility
+        if self.data.get('state', True):
+            for sse in rules:
+                if self.filter_bucket(b, sse):
+                    return True
+        else:
+            if not rules:
```

Comment:
this isn't quite right, the absence of rules doesn't quite mesh with the ability to filter on particular rule expressions with a negation/state=false.

ie. a policy with

```
   crypto: aws:kms
   state: false
```
would ignore buckets with AES256.